core/arch/riscv64/memfuncs.asm

```diff
@@ -44,30 +44,130 @@ START_FILE
  */
         DECLARE_FUNC(memcpy)
 GLOBAL_LABEL(memcpy:)
-/* TODO i#3544: Naive version, optimize it. */
-        mv       t1, ARG1
-        beqz     ARG3, 2f
-1:      lbu      t2, 0(ARG2)
-        addi     ARG2, ARG2, 1
-        sb       t2, 0(t1)
-        addi     t1, t1, 1
-        addi     ARG3, ARG3, -1
-        bnez     ARG3, 1b
-2:      ret
+        addi     a5, x0, 32
```

Comment:
> Refer to musl lib to optimize private memcpy and memset.

If this is based on copyrighted code, that code's license has to be followed.  musl is under the MIT license so its copyright notice would have to be included.  It might be simpler to have someone else do a clean room implementation without having looked at other code.

Reply:
I think it can be considered a clean room implementation, musl does not have a RISC-V assembly version of `memcpy` and `memset`, so it's not "directly" ported code.

Reply:
Let's remove this musl line from the description then to avoid misunderstandings.

Reply:
> Let's remove this musl line from the description then to avoid misunderstandings.

I have modified the description.